pkg/machine/e2e/cp_test.go

```diff
@@ -1,18 +1,210 @@
 package e2e_test
 
 import (
+	"archive/tar"
+	"bytes"
 	"fmt"
+	"io/fs"
 	"os"
+	"path"
 	"path/filepath"
 	"runtime"
+	"time"
 
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
 	. "github.com/onsi/gomega/gexec"
 )
 
-var _ = Describe("podman machine cp", func() {
-	It("all tests", func() {
+var _ = Describe("run cp commands", func() {
+	It("podman cp", func() {
+		var (
+			file = "foo.txt"
+
+			directory = "foo-dir"
+
+			fileInDirectory = "bar.txt"
+
+			stdinFile = "file.txt"
+
+			stdinDirectory = "stdin-dir"
+		)
+
+		sourceDir, err := os.MkdirTemp("", "host-source")
+		Expect(err).ToNot(HaveOccurred())
+
+		destinationDir, err := os.MkdirTemp("", "host-destination")
+		Expect(err).ToNot(HaveOccurred())
```

Comment:
Sorry should have seen this earlier, these are never removed which means you are leaking tmp files.
I suggest using `GinkgoT().TempDir()` instead which removes then by default and also removes the need for error handling.